src/roster.c

```diff
@@ -1826,13 +1870,28 @@ connection_status_changed_cb (GabbleConnection *conn,
         {
           WockyStanza *stanza;
           TpBaseContactList *base = TP_BASE_CONTACT_LIST (self);
+          WockyNode *query = NULL;
+          const gchar *user;
 
           self->priv->cancel_on_disconnect = g_cancellable_new ();
 
           if (tp_base_contact_list_get_download_at_connection (base))
             {
               DEBUG ("requesting roster");
 
+              if (conn->features & GABBLE_CONNECTION_FEATURES_ROSTERVER)
+                {
+                  if (self->priv->rcache == NULL)
+                      self->priv->rcache = roster_cache_dup_shared ();
+                  user = conn_util_get_bare_self_jid (conn);
+                  if (user)
+                      query = roster_cache_get_roster (self->priv->rcache, user);
+                  if (query)
+                    {
+                      process_roster (self, query);
+                      wocky_node_free (query);
+                    }
+                }
               stanza = _gabble_roster_message_new (self, WOCKY_STANZA_SUB_TYPE_GET,
```

Comment:
Shouldn't you avoid requesting the roster after you loaded the cache successfully? Do I miss something?

Reply:
No, we need to request roster updates since last cached version. Cache allows us to save the bandwidth as without any changes we'd get simple <iq type=result/> as confirmation. Otherwise we have two possibilities - server floods updates at us or send full roster (up to server to decide what's better)

Reply:
I see, thanks for the explanation.